tests/info_test.go

```diff
@@ -1,57 +1,62 @@
 package tests
 
 import (
-	"encoding/json"
-	"net/http"
-	"path"
+	"context"
 	"testing"
 
 	// please only use the gateway types in this test
 	// other other tests should use the provider types
-	gwtypes "github.com/openfaas/faas/gateway/types"
+	faasSDK "github.com/openfaas/faas-cli/proxy"
 )
 
 func Test_ProviderInfo(t *testing.T) {
-	gwURL := gatewayUrl(t, path.Join("system", "info"), "")
-	payload, resp := request(t, gwURL, http.MethodGet, nil)
-	if resp.StatusCode != http.StatusOK {
-		t.Fatalf("got status code %d, expected 200", resp.StatusCode)
-	}
+	gwURL := gatewayUrl(t, "", "")
+
+	client := faasSDK.NewClient(&FaaSAuth{}, gwURL, nil, &timeout)
+	systeminfo, err := client.GetSystemInfo(context.Background())
 
-	info := gwtypes.GatewayInfo{}
-	err := json.Unmarshal(payload, &info)
 	if err != nil {
-		t.Logf(string(payload))
-		t.Fatalf("unexpected error unmarshaling provider info: %s", err)
+		t.Fatal(err)
 	}
 
-	if info.Provider == nil {
-		t.Fatal("provider info may not be nil")
+	p, ok := systeminfo["provider"]
+	if !ok {
+		t.Fatal("provider info should be present")
 	}
+	provider := p.(map[string]interface{})
```

Comment:
Why switch to the `map[string]interface{}` here instead of using the gateway type for this?

Reply:
https://github.com/openfaas/faas-cli/issues/796 I have create this issue for this. Once I add this, we can switch back to gateway type.